### ISSUE TITLE
RHEL 8/9 SELinux permission denied with FUSE packer

### ISSUE THREAD

**Issue by PepperJo** [canonical/packer-maas#126]
I tried to use the latest main to create RHEL8 and 9 images but ran into issues: I got tons of permission denied messages in the system log. They all were SELinux related, e.g. MAAS resp. curtin/cloud-init couldn't set hostnames etc.
I reverted back to the commit right before FUSE was introduced and everything seems to work fine now.

**Comment by SK1Y101** [comment]
Hey @PepperJo, could you add a little more information about what you were doing and where it failed?
I've just built a rhel8 and 9 image on the latest commit (1cc01e2), I'm wondering if this has been fixed on your end too, or if we still need to investigate further?

**Comment by PepperJo** [comment]
It looks like an issue with the security context of the files not being preserved correctly (cf. https://fedoraproject.org/wiki/Security_context)

**Comment by SK1Y101** [comment]
Ah, I see: Thank you for the clarification!
Out of interest, what version of MAAS are you running?

Actually, would you mind attaching a log of the failures you encountered? that would be awesome!

**Comment by PepperJo** [comment]
MAAS version 3.3.4

Unfortunately, I didn't save the logs but this one:
```
audit: type=1400 audit(1691611096.562:4): avc: denied { dac_read_search } for pid=2095 comm="systemd-gpt-aut" capability=2...
```
Typically those were accompanied with a permission denied error.

There were multiple of this type leading to cloud-init/curtin not being able to set the hostname, user not being able to read /etc/profile etc.

**Comment by CodeBleu** [comment]
I'm having the same issue when trying to deploy CentOS8-stream and Rocky8.


[root@localhost ~]# uname -a
```
Linux localhost.localdomain 4.18.0-500.el8.x86_64 #1 SMP Wed Jun 28 00:07:07 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
[root@localhost ~]# cat /etc/redhat-release
CentOS Stream release 8
```



[root@localhost ~]# grep -nircP "fail" /var/log | awk -F ":" '{print "File:", $1, "Count:", $2, "Occurrences:", $3}'
```
File: /var/log/dnf.rpm.log Count: 0 Occurrences:
File: /var/log/messages Count: 43 Occurrences:
File: /var/log/secure Count: 0 Occurrences:
File: /var/log/maillog Count: 0 Occurrences:
File: /var/log/kdump.log Count: 1 Occurrences:
File: /var/log/firewalld Count: 0 Occurrences:
File: /var/log/dnf.librepo.log Count: 0 Occurrences:
File: /var/log/cloud-init-output.log Count: 8 Occurrences:
File: /var/log/anaconda/syslog Count: 55 Occurrences:
File: /var/log/anaconda/anaconda.log Count: 1 Occurrences:
File: /var/log/anaconda/dbus.log Count: 0 Occurrences:
File: /var/log/anaconda/lvm.log Count: 2 Occurrences:
File: /var/log/anaconda/storage.log Count: 12 Occurrences:
File: /var/log/anaconda/dnf.librepo.log Count: 0 Occurrences:
File: /var/log/anaconda/ks-script-48vdgkm9.log Count: 0 Occurrences:
File: /var/log/anaconda/hawkey.log Count: 0 Occurrences:
File: /var/log/anaconda/journal.log Count: 70 Occurrences:
File: /var/log/anaconda/program.log Count: 0 Occurrences:
File: /var/log/anaconda/ks-script-x79xk6sb.log Count: 0 Occurrences:
File: /var/log/anaconda/packaging.log Count: 3 Occurrences:
File: /var/log/anaconda/ks-script-1i54rjt9.log Count: 0 Occurrences:
File: /var/log/anaconda/ks-script-zfym_1tp.log Count: 0 Occurrences:
File: /var/log/tuned/tuned.log Count: 0 Occurrences:
File: /var/log/wtmp Count: 0 Occurrences:
File: /var/log/dnf.log Count: 0 Occurrences:
File: /var/log/spooler Count: 0 Occurrences:
File: /var/log/btmp Count: 0 Occurrences:
File: /var/log/lastlog Count: 0 Occurrences:
File: /var/log/cron Count: 0 Occurrences:
File: /var/log/cloud-init.log Count: 24 Occurrences:
```


[root@localhost ~]# grep -nircP "denied" /var/log | awk -F ":" '{print "File:", $1, "Count:", $2, "Occurrences:", $3}'
```
File: /var/log/dnf.rpm.log Count: 0 Occurrences:
File: /var/log/messages Count: 51 Occurrences:
File: /var/log/secure Count: 1 Occurrences:
File: /var/log/maillog Count: 0 Occurrences:
File: /var/log/kdump.log Count: 0 Occurrences:
File: /var/log/firewalld Count: 0 Occurrences:
File: /var/log/dnf.librepo.log Count: 0 Occurrences:
File: /var/log/cloud-init-output.log Count: 0 Occurrences:
File: /var/log/anaconda/syslog Count: 0 Occurrences:
File: /var/log/anaconda/anaconda.log Count: 0 Occurrences:
File: /var/log/anaconda/dbus.log Count: 0 Occurrences:
File: /var/log/anaconda/lvm.log Count: 0 Occurrences:
File: /var/log/anaconda/storage.log Count: 0 Occurrences:
File: /var/log/anaconda/dnf.librepo.log Count: 0 Occurrences:
File: /var/log/anaconda/ks-script-48vdgkm9.log Count: 0 Occurrences:
File: /var/log/anaconda/hawkey.log Count: 0 Occurrences:
File: /var/log/anaconda/journal.log Count: 0 Occurrences:
File: /var/log/anaconda/program.log Count: 0 Occurrences:
File: /var/log/anaconda/ks-script-x79xk6sb.log Count: 0 Occurrences:
File: /var/log/anaconda/packaging.log Count: 0 Occurrences:
File: /var/log/anaconda/ks-script-1i54rjt9.log Count: 0 Occurrences:
File: /var/log/anaconda/ks-script-zfym_1tp.log Count: 0 Occurrences:
File: /var/log/tuned/tuned.log Count: 0 Occurrences:
File: /var/log/wtmp Count: 0 Occurrences:
File: /var/log/dnf.log Count: 0 Occurrences:
File: /var/log/spooler Count: 0 Occurrences:
File: /var/log/btmp Count: 0 Occurrences:
File: /var/log/lastlog Count: 0 Occurrences:
File: /var/log/cron Count: 0 Occurrences:
File: /var/log/cloud-init.log Count: 1 Occurrences:
```

[root@localhost ~]# grep -iP 'denied' /var/log/messages
```
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.035:4): avc:  denied  { dac_read_search } for  pid=625 comm="systemd-gpt-aut" capability=2  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.035:5): avc:  denied  { dac_override } for  pid=625 comm="systemd-gpt-aut" capability=1  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.035:6): avc:  denied  { dac_read_search } for  pid=625 comm="systemd-gpt-aut" capability=2  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.035:7): avc:  denied  { dac_override } for  pid=625 comm="systemd-gpt-aut" capability=1  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.093:8): avc:  denied  { dac_read_search } for  pid=625 comm="systemd-gpt-aut" capability=2  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.093:9): avc:  denied  { dac_override } for  pid=625 comm="systemd-gpt-aut" capability=1  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.093:10): avc:  denied  { dac_read_search } for  pid=625 comm="systemd-gpt-aut" capability=2  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.346:12): avc:  denied  { dac_read_search } for  pid=625 comm="systemd-gpt-aut" capability=2  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.346:13): avc:  denied  { dac_override } for  pid=625 comm="systemd-gpt-aut" capability=1  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.346:14): avc:  denied  { dac_read_search } for  pid=625 comm="systemd-gpt-aut" capability=2  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.346:15): avc:  denied  { dac_override } for  pid=625 comm="systemd-gpt-aut" capability=1  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.365:16): avc:  denied  { dac_read_search } for  pid=625 comm="systemd-gpt-aut" capability=2  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.365:17): avc:  denied  { dac_override } for  pid=625 comm="systemd-gpt-aut" capability=1  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.381:18): avc:  denied  { dac_read_search } for  pid=625 comm="systemd-gpt-aut" capability=2  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.381:19): avc:  denied  { dac_override } for  pid=625 comm="systemd-gpt-aut" capability=1  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.448:20): avc:  denied  { dac_read_search } for  pid=625 comm="systemd-gpt-aut" capability=2  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:55 localhost kernel: audit: type=1400 audit(1692717173.448:21): avc:  denied  { dac_override } for  pid=625 comm="systemd-gpt-aut" capability=1  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0
Aug 22 15:12:57 localhost auditd[813]: Error opening config file (Permission denied)
Aug 22 15:12:57 localhost auditd[850]: Error opening config file (Permission denied)
Aug 22 15:12:58 localhost dbus-daemon[846]: Cannot setup inotify for '/etc/dbus-1/system.d'; error 'Permission denied'
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.646:364): avc:  denied  { dac_read_search } for  pid=901 comm="auditd" capability=2  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.646:365): avc:  denied  { dac_override } for  pid=901 comm="auditd" capability=1  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.646:366): avc:  denied  { dac_read_search } for  pid=901 comm="auditd" capability=2  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.646:367): avc:  denied  { dac_override } for  pid=901 comm="auditd" capability=1  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.648:368): avc:  denied  { dac_read_search } for  pid=901 comm="auditd" capability=2  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.648:369): avc:  denied  { dac_override } for  pid=901 comm="auditd" capability=1  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.648:370): avc:  denied  { dac_read_search } for  pid=901 comm="auditd" capability=2  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.648:371): avc:  denied  { dac_override } for  pid=901 comm="auditd" capability=1  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:58 localhost auditd[901]: Error opening config file (Permission denied)
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.683:372): avc:  denied  { dac_read_search } for  pid=901 comm="auditd" capability=2  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:58 localhost kernel: audit: type=1400 audit(1692717178.683:373): avc:  denied  { dac_override } for  pid=901 comm="auditd" capability=1  scontext=system_u:system_r:auditd_t:s0 tcontext=system_u:system_r:auditd_t:s0 tclass=capability permissive=0
Aug 22 15:12:59 localhost auditd[903]: Error opening config file (Permission denied)
Aug 22 15:12:59 localhost auditd[907]: Error opening config file (Permission denied)
Aug 22 15:13:01 localhost systemd-hostnamed[994]: Failed to read hostname and machine information: Permission denied
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.492:428): avc:  denied  { dac_read_search } for  pid=1341 comm="ip" capability=2  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.492:429): avc:  denied  { dac_override } for  pid=1341 comm="ip" capability=1  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.492:430): avc:  denied  { dac_read_search } for  pid=1341 comm="ip" capability=2  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.492:431): avc:  denied  { dac_override } for  pid=1341 comm="ip" capability=1  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.493:432): avc:  denied  { dac_read_search } for  pid=1341 comm="ip" capability=2  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.493:433): avc:  denied  { dac_override } for  pid=1341 comm="ip" capability=1  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.493:434): avc:  denied  { dac_read_search } for  pid=1341 comm="ip" capability=2  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.493:435): avc:  denied  { dac_override } for  pid=1341 comm="ip" capability=1  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.497:436): avc:  denied  { dac_read_search } for  pid=1342 comm="ip" capability=2  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:08 localhost kernel: audit: type=1400 audit(1692717188.497:437): avc:  denied  { dac_override } for  pid=1342 comm="ip" capability=1  scontext=system_u:system_r:ifconfig_t:s0 tcontext=system_u:system_r:ifconfig_t:s0 tclass=capability permissive=0
Aug 22 15:13:09 localhost systemd-hostnamed[1351]: Failed to read hostname and machine information: Permission denied
Aug 22 15:13:10 localhost systemd-hostnamed[1361]: Failed to read hostname and machine information: Permission denied
Aug 22 15:13:20 localhost kdumpctl[5798]: ldconfig: Can't create temporary cache file /etc/ld.so.cache~: Permission denied
Aug 22 15:13:20 localhost kernel: audit: type=1400 audit(1692717200.906:476): avc:  denied  { dac_override } for  pid=5798 comm="ldconfig" capability=1  scontext=system_u:system_r:ldconfig_t:s0 tcontext=system_u:system_r:ldconfig_t:s0 tclass=capability permissive=0
Aug 22 15:14:29 localhost systemd[6025]: Failed to open configuration file '/etc/systemd/user.conf': Permission denied
Aug 22 15:14:29 localhost systemd[6025]: Failed to create timezone change event source: Permission denied
Aug 22 15:14:29 localhost systemd[6025]: Failed to allocate manager object: Permission denied
```

[root@localhost ~]# grep -iP 'failed' /var/log/messages
```
Aug 22 15:12:49 localhost kernel: pci 0000:02:00.0: BAR 7: failed to assign [mem size 0x00100000 64bit]
Aug 22 15:12:49 localhost kernel: pci 0000:02:00.0: BAR 10: failed to assign [mem size 0x00100000 64bit]
Aug 22 15:12:49 localhost kernel: pci 0000:02:00.1: BAR 7: failed to assign [mem size 0x00100000 64bit]
Aug 22 15:12:49 localhost kernel: pci 0000:02:00.1: BAR 10: failed to assign [mem size 0x00100000 64bit]
Aug 22 15:12:57 localhost systemd[1]: auditd.service: Failed with result 'exit-code'.
Aug 22 15:12:57 localhost systemd[1]: Failed to start Security Auditing Service.
Aug 22 15:12:58 localhost systemd[1]: auditd.service: Failed with result 'exit-code'.
Aug 22 15:12:58 localhost systemd[1]: Failed to start Security Auditing Service.
Aug 22 15:12:58 localhost systemd[1]: auditd.service: Failed with result 'exit-code'.
Aug 22 15:12:58 localhost systemd[1]: Failed to start Security Auditing Service.
Aug 22 15:12:59 localhost systemd[1]: auditd.service: Failed with result 'exit-code'.
Aug 22 15:12:59 localhost systemd[1]: Failed to start Security Auditing Service.
Aug 22 15:12:59 localhost systemd[1]: auditd.service: Failed with result 'exit-code'.
Aug 22 15:12:59 localhost systemd[1]: Failed to start Security Auditing Service.
Aug 22 15:13:00 localhost systemd[1]: auditd.service: Failed with result 'exit-code'.
Aug 22 15:13:00 localhost systemd[1]: Failed to start Security Auditing Service.
Aug 22 15:13:00 localhost cloud-init[933]: 2023-08-22 15:13:00,609 - handlers.py[WARNING]: Failed posting event: {"name": "init-local/check-cache", "description": "attempting to read from cache [check]", "event_type": "start", "origin": "cloudinit", "timestamp": 1692717180.476108}. This was caused by: HTTPConnectionPool(host='10.120.0.51', port=5248): Max retries exceeded with url: /MAAS/metadata/status/f3m4rx (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f34dbc544e0>: Failed to establish a new connection: [Errno 101] Network is unreachable',))
Aug 22 15:13:00 localhost cloud-init[933]: 2023-08-22 15:13:00,611 - handlers.py[WARNING]: Failed posting event: {"name": "init-local/check-cache", "description": "no cache found", "event_type": "finish", "origin": "cloudinit", "timestamp": 1692717180.4766953, "result": "SUCCESS"}. This was caused by: HTTPConnectionPool(host='10.120.0.51', port=5248): Max retries exceeded with url: /MAAS/metadata/status/f3m4rx (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f34dbc54d68>: Failed to establish a new connection: [Errno 101] Network is unreachable',))
Aug 22 15:13:00 localhost cloud-init[933]: 2023-08-22 15:13:00,777 - handlers.py[WARNING]: Failed posting event: {"name": "init-local", "description": "searching for local datasources", "event_type": "finish", "origin": "cloudinit", "timestamp": 1692717180.7738776, "result": "SUCCESS"}. This was caused by: HTTPConnectionPool(host='10.120.0.51', port=5248): Max retries exceeded with url: /MAAS/metadata/status/f3m4rx (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f34dcdf7438>: Failed to establish a new connection: [Errno 101] Network is unreachable',))
Aug 22 15:13:01 localhost systemd-hostnamed[994]: Failed to read hostname and machine information: Permission denied
Aug 22 15:13:01 localhost systemd[1]: systemd-hostnamed.service: Failed with result 'exit-code'.
Aug 22 15:13:07 localhost dbus-daemon[846]: [system] Activation via systemd failed for unit 'dbus-org.freedesktop.resolve1.service': Unit dbus-org.freedesktop.resolve1.service not found.
Aug 22 15:13:09 localhost cloud-init[1339]: 2023-08-22 15:13:09,100 - util.py[WARNING]: Running interface command ['nmcli', 'connection', 'up', 'ifname', 'eno1'] failed
Aug 22 15:13:09 localhost systemd-hostnamed[1351]: Failed to read hostname and machine information: Permission denied
Aug 22 15:13:09 localhost cloud-init[1339]: 2023-08-22 15:13:09,487 - util.py[WARNING]: Failed to set the hostname to large-guinea.maas (large-guinea)
Aug 22 15:13:09 localhost systemd[1]: systemd-hostnamed.service: Failed with result 'exit-code'.
Aug 22 15:13:10 localhost systemd-hostnamed[1361]: Failed to read hostname and machine information: Permission denied
Aug 22 15:13:10 localhost cloud-init[1339]: 2023-08-22 15:13:10,247 - util.py[WARNING]: Failed to set the hostname to large-guinea.maas (large-guinea)
Aug 22 15:13:10 localhost cloud-init[1339]: 2023-08-22 15:13:10,248 - util.py[WARNING]: Running module set_hostname (<module 'cloudinit.config.cc_set_hostname' from '/usr/lib/python3.6/site-packages/cloudinit/config/cc_set_hostname.py'>) failed
Aug 22 15:13:10 localhost systemd[1]: systemd-hostnamed.service: Failed with result 'exit-code'.
Aug 22 15:13:12 localhost systemd[1]: cloud-init.service: Failed with result 'exit-code'.
Aug 22 15:13:12 localhost systemd[1]: Failed to start Initial cloud-init job (metadata service crawler).
Aug 22 15:14:29 localhost systemd[6025]: Failed to open configuration file '/etc/systemd/user.conf': Permission denied
Aug 22 15:14:29 localhost systemd[6025]: Failed to create timezone change event source: Permission denied
Aug 22 15:14:29 localhost systemd[6025]: Failed to allocate manager object: Permission denied
Aug 22 15:14:29 localhost systemd[1]: user@1000.service: Failed with result 'protocol'.
Aug 22 15:14:29 localhost systemd[1]: Failed to start User Manager for UID 1000.
```

**Comment by CodeBleu** [comment]
I had tried disabling `selinux`  before as well, but still got the same errors and would also have issues with the packer build hanging up on the `fusermount -u` part.

I can confirm that checking out the commit before FUSE stuff ( #0057b ) with `git checkout 0057b` and putting `selinux --disabled` in the centos8-stream.ks file, I was able to run the packer build with no issues and the deploy in MaaS was able to deploy with no issues.  It was able to set the hostname and no other permission denied errors.  One other observation, was the `cloud-init status` had came back `status: failed` before, but this time it returned `status: done`
